stellarisdashboard/parsing/save_parser.py

```diff
@@ -261,26 +262,21 @@ def parse_save(filename) -> Dict[str, Any]:
     :param filename: Path to a .sav file
     :return: The gamestate dictionary
     """
-    gamestate = None
-    parser = SaveFileParser()
-    max_attempts = 3
-    for attempt in range(max_attempts):
-        try:
-            gamestate = parser.parse_save(filename)
-            break
-        except KeyboardInterrupt:
-            raise
-        except zipfile.BadZipFile as e:
-            remaining = max_attempts - attempt - 1
-            if not remaining:
-                raise
-            delay_seconds = 0.5
-            logger.info(
-                f"Encountered BadZipFile error {e}. Next attempt in {delay_seconds} seconds, {remaining} attempts remaining."
-            )
-            time.sleep(delay_seconds)
-    return gamestate
+    import rust_parser
```

Comment:
Ah, question - where do you plan for `rust_parser.so` to live? I put it at `stellarisdashboard/parsing/rust_parser.so` but I imagine (without testing) that the import statement would work as-is if you put it higher in the tree than that.

Reply:
yes what I did locally was to open the rust_parser directory and run `maturin develop -r` which compiles the rust part with optimizations and installs the extension in the virtualenv. I haven't given it much thought yet how to integrate it in the regular build and release yet, so far I was just happy to get it working 😄

Reply:
another part I'm not quite happy with yet is how the data is passed back from rust to python, right now it's serialized to json first, but I believe it should be possible to directly assemble the Python dict from within rust, I expect it would be much more efficient, I'll give that another try that when I have some time again in the next 1-2 weeks